cohortextractor2/query_engines/base_sql.py

```diff
@@ -397,25 +340,18 @@ def get_value_expression(self, value):
         tables = ()
         value_expr = value
         if self.is_category_node(value):
-            category_definitions = value.definitions.copy()
-            all_category_referenced_nodes = (
-                self.list_parent_nodes_from_category_definitions(
-                    category_definitions.values()
-                )
-            )
-            tables = tuple(
-                self.output_group_tables[self.get_output_group(query_node)]
-                for query_node in all_category_referenced_nodes
-            )
```

Comment:
This seems like the first step on the way to getting rid of the two parallel data structures, is that right?

Reply:
It is, yes!